<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Revisión | Intro RNA-seq LCG-UNAM 2021</title>
  <meta name="description" content="Book from first module of Bioinformática 2" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Revisión | Intro RNA-seq LCG-UNAM 2021" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Book from first module of Bioinformática 2" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Revisión | Intro RNA-seq LCG-UNAM 2021" />
  
  <meta name="twitter:description" content="Book from first module of Bioinformática 2" />
  

<meta name="author" content="Leonardo Collado-Torres, Alfredo Varela Vega" />


<meta name="date" content="2021-02-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="específicaciones-del-proyecto.html"/>
<link rel="next" href="spatiallibd.html"/>
<script src="assets/jquery-2.2.3/jquery.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="assets/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a><ul>
<li class="chapter" data-level="0.0.1" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.0.1</b> Software</a></li>
<li class="chapter" data-level="0.0.2" data-path="index.html"><a href="index.html#anotación"><i class="fa fa-check"></i><b>0.0.2</b> Anotación</a></li>
<li class="chapter" data-level="0.0.3" data-path="index.html"><a href="index.html#experiment-data"><i class="fa fa-check"></i><b>0.0.3</b> Experiment Data</a></li>
<li class="chapter" data-level="0.1" data-path="workflows.html"><a href="workflows.html"><i class="fa fa-check"></i><b>0.1</b> Workflows</a></li>
<li class="chapter" data-level="0.2" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html"><i class="fa fa-check"></i><b>0.2</b> SummarizedExperiment</a></li>
<li class="chapter" data-level="0.3" data-path="para-encontrar-paquetes.html"><a href="para-encontrar-paquetes.html"><i class="fa fa-check"></i><b>0.3</b> Para encontrar paquetes</a></li>
<li class="chapter" data-level="0.4" data-path="dos-versiones-de-bioconductor-release-y-devel.html"><a href="dos-versiones-de-bioconductor-release-y-devel.html"><i class="fa fa-check"></i><b>0.4</b> Dos versiones de Bioconductor (Release y Devel)</a></li>
<li class="chapter" data-level="0.5" data-path="actulaizaciones-de-r-y-bioconductor.html"><a href="actulaizaciones-de-r-y-bioconductor.html"><i class="fa fa-check"></i><b>0.5</b> Actulaizaciones de R y Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="paquetes-en-equipo.html"><a href="paquetes-en-equipo.html"><i class="fa fa-check"></i><b>1</b> Paquetes en Equipo</a></li>
<li class="chapter" data-level="2" data-path="objetos.html"><a href="objetos.html"><i class="fa fa-check"></i><b>2</b> Objetos</a><ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-workflow-gene-level-exploratory-analysis-and-differential-expression.html"><a href="rna-seq-workflow-gene-level-exploratory-analysis-and-differential-expression.html"><i class="fa fa-check"></i><b>2.1</b> RNA-seq workflow: gene-level exploratory analysis and differential expression</a></li>
<li class="chapter" data-level="2.2" data-path="summarized-experiment-tipo-de-objeto.html"><a href="summarized-experiment-tipo-de-objeto.html"><i class="fa fa-check"></i><b>2.2</b> Summarized Experiment (tipo de objeto)</a></li>
<li class="chapter" data-level="2.3" data-path="genomic-ranges-tipo-de-objeto.html"><a href="genomic-ranges-tipo-de-objeto.html"><i class="fa fa-check"></i><b>2.3</b> Genomic Ranges (tipo de objeto)</a></li>
<li class="chapter" data-level="2.4" data-path="ejercicio.html"><a href="ejercicio.html"><i class="fa fa-check"></i><b>2.4</b> Ejercicio</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ejercicio-2-realizar-el-heatmap.html"><a href="ejercicio-2-realizar-el-heatmap.html"><i class="fa fa-check"></i><b>3</b> Ejercicio 2 Realizar el heatmap</a></li>
<li class="chapter" data-level="4" data-path="proyecto-recount.html"><a href="proyecto-recount.html"><i class="fa fa-check"></i><b>4</b> Proyecto Recount</a></li>
<li class="chapter" data-level="5" data-path="modelos.html"><a href="modelos.html"><i class="fa fa-check"></i><b>5</b> Modelos</a><ul>
<li class="chapter" data-level="5.1" data-path="exploremodelmatrix.html"><a href="exploremodelmatrix.html"><i class="fa fa-check"></i><b>5.1</b> ExploreModelMatrix</a><ul>
<li class="chapter" data-level="5.1.1" data-path="exploremodelmatrix.html"><a href="exploremodelmatrix.html#ejemplo-1"><i class="fa fa-check"></i><b>5.1.1</b> Ejemplo 1</a></li>
<li class="chapter" data-level="5.1.2" data-path="exploremodelmatrix.html"><a href="exploremodelmatrix.html#ejemplo-2"><i class="fa fa-check"></i><b>5.1.2</b> Ejemplo 2</a></li>
<li class="chapter" data-level="5.1.3" data-path="exploremodelmatrix.html"><a href="exploremodelmatrix.html#ejercicio-1"><i class="fa fa-check"></i><b>5.1.3</b> Ejercicio</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="datos-de-srp045638.html"><a href="datos-de-srp045638.html"><i class="fa fa-check"></i><b>5.2</b> Datos de SRP045638</a></li>
<li class="chapter" data-level="5.3" data-path="normalización-de-datos.html"><a href="normalización-de-datos.html"><i class="fa fa-check"></i><b>5.3</b> Normalización de datos</a></li>
<li class="chapter" data-level="5.4" data-path="expresión-diferencial.html"><a href="expresión-diferencial.html"><i class="fa fa-check"></i><b>5.4</b> Expresión diferencial</a></li>
<li class="chapter" data-level="5.5" data-path="visualizando-genes-de.html"><a href="visualizando-genes-de.html"><i class="fa fa-check"></i><b>5.5</b> Visualizando genes DE</a></li>
<li class="chapter" data-level="5.6" data-path="ejercicio-2.html"><a href="ejercicio-2.html"><i class="fa fa-check"></i><b>5.6</b> Ejercicio</a></li>
<li class="chapter" data-level="5.7" data-path="comunidad.html"><a href="comunidad.html"><i class="fa fa-check"></i><b>5.7</b> Comunidad</a></li>
<li class="chapter" data-level="5.8" data-path="ejercicio-respuesta.html"><a href="ejercicio-respuesta.html"><i class="fa fa-check"></i><b>5.8</b> Ejercicio: respuesta</a></li>
<li class="chapter" data-level="5.9" data-path="específicaciones-del-proyecto.html"><a href="específicaciones-del-proyecto.html"><i class="fa fa-check"></i><b>5.9</b> Específicaciones del proyecto</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="revisión.html"><a href="revisión.html"><i class="fa fa-check"></i><b>6</b> Revisión</a></li>
<li class="chapter" data-level="7" data-path="spatiallibd.html"><a href="spatiallibd.html"><i class="fa fa-check"></i><b>7</b> spatialLIBD</a><ul>
<li class="chapter" data-level="7.1" data-path="comunidad-1.html"><a href="comunidad-1.html"><i class="fa fa-check"></i><b>7.1</b> Comunidad</a></li>
<li class="chapter" data-level="7.2" data-path="de-ustedes.html"><a href="de-ustedes.html"><i class="fa fa-check"></i><b>7.2</b> De ustedes</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Intro RNA-seq LCG-UNAM 2021</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="revisión" class="section level1">
<h1><span class="header-section-number">6</span> Revisión</h1>
<ul>
<li>¿Debemos explorar las relaciones entre nuestras variables?</li>
</ul>
<p>Sí y depende de eso podemos elegir el modelo
* ¿Por qué usamos el paquete <code>edgeR</code>?
Sirve para normalizar los datos
Nos permite eliminar el composition bias</p>
<ul>
<li>¿Por qué es importante el argumento <code>sort.by</code> en <code>topTable()</code>?
queriamos preservar el orden original para que los renglones del objeto rse tengan el mismo orden de la tabla</li>
<li>¿Por qué es importante el argumento <code>coef</code> en <code>topTable()</code>?
Es el coeficiente de interés lo que corresponde a model.matrix con la que creamos la matriz del diseño estadístico</li>
</ul>
<p>Usemos los datos de <a href="http://research.libd.org/SPEAQeasy-example/bootcamp_intro" class="uri">http://research.libd.org/SPEAQeasy-example/bootcamp_intro</a></p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="revisión.html#cb176-1"></a>speaqeasy_data &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(), <span class="st">&quot;rse_speaqeasy.RData&quot;</span>)</span>
<span id="cb176-2"><a href="revisión.html#cb176-2"></a><span class="kw">download.file</span>(<span class="st">&quot;https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/rse_speaqeasy.RData?raw=true&quot;</span>, speaqeasy_data, <span class="dt">mode =</span> <span class="st">&quot;wb&quot;</span>)</span>
<span id="cb176-3"><a href="revisión.html#cb176-3"></a><span class="kw">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</span>
<span id="cb176-4"><a href="revisión.html#cb176-4"></a><span class="co"># cargamos los datos y con verbose R nos avisa cuales son los nombres de los objetos </span></span>
<span id="cb176-5"><a href="revisión.html#cb176-5"></a><span class="kw">load</span>(speaqeasy_data, <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## Loading objects:
##   rse_gene</code></pre>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="revisión.html#cb178-1"></a>rse_gene</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 60609 40 
## metadata(0):
## assays(1): counts
## rownames(60609): ENSG00000223972.5 ENSG00000227232.5 ... ENSG00000210195.2 ENSG00000210196.2
## rowData names(10): Length gencodeID ... NumTx gencodeTx
## colnames(40): R13896_H7JKMBBXX R13903_HCTYLBBXX ... R15120_HFY2MBBXX R15134_HFFGHBBXX
## colData names(67): SAMPLE_ID FQCbasicStats ... AgeDeath BrNum</code></pre>
<ul>
<li>¿Cuantos genes y muestras tenemos en estos datos?
genes tenemos 60606
muestras 40
¿Cuantas variables de info en nuestras muestras?
colData names (67)</li>
<li>¿Hay diferencias en <code>totalAssignedGene</code> o <code>mitoRate</code> entre los grupos de diagnosis (<code>PrimaryDx</code>)?</li>
</ul>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="revisión.html#cb180-1"></a><span class="co"># isee o ggplot2, scater</span></span>
<span id="cb180-2"><a href="revisión.html#cb180-2"></a><span class="kw">colData</span>(rse_gene)<span class="op">$</span>names</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="revisión.html#cb182-1"></a>rse_gene<span class="op">$</span>totalAssignedGene <span class="op">==</span><span class="st"> </span>rse_gene<span class="op">$</span>mitoRate</span></code></pre></div>
<pre><code>##  [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
## [20] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
## [39] FALSE FALSE</code></pre>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="revisión.html#cb184-1"></a>rse_gene<span class="op">$</span>PrimaryDx</span></code></pre></div>
<pre><code>##  [1] Bipolar Bipolar Bipolar Bipolar Control Control Control Control Control Control Control Control Control Bipolar
## [15] Bipolar Bipolar Control Control Control Bipolar Bipolar Control Bipolar Bipolar Bipolar Bipolar Control Control
## [29] Control Bipolar Bipolar Bipolar Control Control Bipolar Bipolar Bipolar Bipolar Control Control
## Levels: Bipolar Control Other</code></pre>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="revisión.html#cb186-1"></a><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb186-2"><a href="revisión.html#cb186-2"></a><span class="kw">ggplot</span>(<span class="kw">as.data.frame</span>(<span class="kw">colData</span>(rse_gene)), <span class="kw">aes</span>(<span class="dt">y =</span> mitoRate, <span class="dt">x =</span> PrimaryDx)) <span class="op">+</span></span>
<span id="cb186-3"><a href="revisión.html#cb186-3"></a><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span></span>
<span id="cb186-4"><a href="revisión.html#cb186-4"></a><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dt">base_size =</span> <span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb186-5"><a href="revisión.html#cb186-5"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;mitoRate&quot;</span>) <span class="op">+</span></span>
<span id="cb186-6"><a href="revisión.html#cb186-6"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PrimaryDx&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="revisión.html#cb187-1"></a><span class="kw">ggplot</span>(<span class="kw">as.data.frame</span>(<span class="kw">colData</span>(rse_gene)), <span class="kw">aes</span>(<span class="dt">y =</span> totalAssignedGene, <span class="dt">x =</span> PrimaryDx)) <span class="op">+</span></span>
<span id="cb187-2"><a href="revisión.html#cb187-2"></a><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span></span>
<span id="cb187-3"><a href="revisión.html#cb187-3"></a><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dt">base_size =</span> <span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb187-4"><a href="revisión.html#cb187-4"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;totalAssignedGene&quot;</span>) <span class="op">+</span></span>
<span id="cb187-5"><a href="revisión.html#cb187-5"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PrimaryDx&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<ul>
<li>Grafica la expresión de <em>SNAP25</em> para cada grupo de diagnosis.</li>
</ul>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="revisión.html#cb188-1"></a><span class="co"># Es un gen marcador de neuronas, no debería haber diferencia ggplot2, scater o isee</span></span></code></pre></div>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="revisión.html#cb189-1"></a>rse_gene</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 60609 40 
## metadata(0):
## assays(1): counts
## rownames(60609): ENSG00000223972.5 ENSG00000227232.5 ... ENSG00000210195.2 ENSG00000210196.2
## rowData names(10): Length gencodeID ... NumTx gencodeTx
## colnames(40): R13896_H7JKMBBXX R13903_HCTYLBBXX ... R15120_HFY2MBBXX R15134_HFFGHBBXX
## colData names(67): SAMPLE_ID FQCbasicStats ... AgeDeath BrNum</code></pre>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="revisión.html#cb191-1"></a><span class="kw">rowData</span>(rse_gene)<span class="op">$</span>Symbol<span class="op">==</span><span class="st"> &quot;SNAP25&quot;</span></span></code></pre></div>
<pre><code>##    [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##   [20] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##   [39] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##   [58] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##   [77] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##   [96] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [115] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [134] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [153] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [172] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [191] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [210] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [229] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [248] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [267] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [286] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [305] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [324] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [343] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [362] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [381] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [400] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [419] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [438] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [457] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [476] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [495] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [514] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [533] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [552] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [571] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [590] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [609] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [628] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [647] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [666] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [685] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [704] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [723] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [742] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [761] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [780] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [799] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [818] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [837] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [856] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [875] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [894] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [913] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [932] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [951] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [970] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [989] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
##  [ reached getOption(&quot;max.print&quot;) -- omitted 59609 entries ]</code></pre>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="revisión.html#cb193-1"></a>index_snap&lt;-<span class="kw">match</span>(<span class="kw">c</span>(<span class="st">&quot;SNAP25&quot;</span>), <span class="kw">rowData</span>(rse_gene)<span class="op">$</span>Symbol)</span>
<span id="cb193-2"><a href="revisión.html#cb193-2"></a></span>
<span id="cb193-3"><a href="revisión.html#cb193-3"></a>rse_gene[index_snap]</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 1 40 
## metadata(0):
## assays(1): counts
## rownames(1): ENSG00000132639.12
## rowData names(10): Length gencodeID ... NumTx gencodeTx
## colnames(40): R13896_H7JKMBBXX R13903_HCTYLBBXX ... R15120_HFY2MBBXX R15134_HFFGHBBXX
## colData names(67): SAMPLE_ID FQCbasicStats ... AgeDeath BrNum</code></pre>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="revisión.html#cb195-1"></a><span class="kw">library</span>(<span class="st">&quot;iSEE&quot;</span>)</span>
<span id="cb195-2"><a href="revisión.html#cb195-2"></a>iSEE<span class="op">::</span><span class="kw">iSEE</span>(rse_gene[index_snap])</span></code></pre></div>
<p><div style="width: 100% ; height: 400px ; text-align: center; box-sizing: border-box; -moz-box-sizing: border-box; -webkit-box-sizing: border-box;" class="muted well">Shiny applications not supported in static R Markdown documents</div></p>
<ul>
<li>Sugiere un modelo estadistico que podríamos usar en una análisis de expresión diferencial. Verifica que si sea un modelo <em>full rank</em>. ¿Cúal sería el o los coeficientes de interés?
Si todas las columnas son linealmente independientes a las otras. Si no es full rank no se puede usar para el analisis de expresión diferencial</li>
</ul>
<p>Utilizamos el modelo de expresión diferencial</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="revisión.html#cb196-1"></a><span class="co"># explormodelmatrix </span></span>
<span id="cb196-2"><a href="revisión.html#cb196-2"></a></span>
<span id="cb196-3"><a href="revisión.html#cb196-3"></a>mod &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="st"> </span>PrimaryDx <span class="op">+</span><span class="st"> </span>totalAssignedGene <span class="op">+</span><span class="st"> </span>mitoRate <span class="op">+</span><span class="st"> </span>numReads,</span>
<span id="cb196-4"><a href="revisión.html#cb196-4"></a>    <span class="dt">data =</span> <span class="kw">colData</span>(rse_gene)</span>
<span id="cb196-5"><a href="revisión.html#cb196-5"></a>)</span>
<span id="cb196-6"><a href="revisión.html#cb196-6"></a><span class="kw">colnames</span>(mod)</span></code></pre></div>
<pre><code>## [1] &quot;(Intercept)&quot;       &quot;PrimaryDxControl&quot;  &quot;PrimaryDxOther&quot;    &quot;totalAssignedGene&quot; &quot;mitoRate&quot;         
## [6] &quot;numReads&quot;</code></pre>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="revisión.html#cb198-1"></a><span class="kw">library</span>(<span class="st">&quot;edgeR&quot;</span>) <span class="co"># BiocManager::install(&quot;edgeR&quot;, update = FALSE)</span></span>
<span id="cb198-2"><a href="revisión.html#cb198-2"></a>dge &lt;-<span class="st"> </span><span class="kw">DGEList</span>(</span>
<span id="cb198-3"><a href="revisión.html#cb198-3"></a>    <span class="dt">counts =</span> <span class="kw">assay</span>(rse_gene, <span class="st">&quot;counts&quot;</span>),</span>
<span id="cb198-4"><a href="revisión.html#cb198-4"></a>    <span class="dt">genes =</span> <span class="kw">rowData</span>(rse_gene)</span>
<span id="cb198-5"><a href="revisión.html#cb198-5"></a>)</span>
<span id="cb198-6"><a href="revisión.html#cb198-6"></a>dge &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(dge)</span></code></pre></div>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="revisión.html#cb199-1"></a><span class="kw">library</span>(<span class="st">&quot;limma&quot;</span>)</span>
<span id="cb199-2"><a href="revisión.html#cb199-2"></a>vGene&lt;-<span class="st"> </span><span class="kw">voom</span>(dge,mod,<span class="dt">plot=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## Coefficients not estimable: PrimaryDxOther</code></pre>
<pre><code>## Warning: Partial NA coefficients for 60609 probe(s)</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>¿Quieres más datos? Tenemos muchos en LIBD incluyendo <a href="http://eqtl.brainseq.org/phase2/" class="uri">http://eqtl.brainseq.org/phase2/</a>.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="específicaciones-del-proyecto.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatiallibd.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/lunr.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
